packages/cw721/src/execute.rs

```diff
@@ -0,0 +1,810 @@
+    {
+        return Ok(());
+    }
+
+    // operator can send
+    let config = Cw721Config::<Empty, Empty, Empty, Empty, Empty>::default();
+    let op = config
+        .operators
+        // has token owner approved/gave grant to sender for full control over owner's NFTs?
+        .may_load(deps.storage, (&token.owner, &info.sender))?;
+
+    match op {
+        Some(ex) => {
+            if ex.is_expired(&env.block) {
+                Err(Cw721ContractError::Ownership(OwnershipError::NotOwner))
+            } else {
+                Ok(())
+            }
+        }
+        None => Err(Cw721ContractError::Ownership(OwnershipError::NotOwner)),
+    }
+}
+
+// ------- migrate -------
+pub fn migrate_version(
+    storage: &mut dyn Storage,
+    contradct_name: &str,
+    contract_version: &str,
+    response: Response,
+) -> StdResult<Response> {
+    let response = response
+        .add_attribute("from_version", cw2::get_contract_version(storage)?.version)
+        .add_attribute("to_version", contract_version);
+
+    // update contract version
+    cw2::set_contract_version(storage, contradct_name, contract_version)?;
+    Ok(response)
+}
+
+pub fn migrate_creator(
+    storage: &mut dyn Storage,
+    api: &dyn Api,
+    _env: &Env,
+    msg: &Cw721MigrateMsg,
+    response: Response,
+) -> StdResult<Response> {
+    match msg {
+        Cw721MigrateMsg::WithUpdate { creator, .. } => {
+            if let Some(creator) = creator {
+                CREATOR.initialize_owner(storage, api, Some(creator.as_str()))?;
+                return Ok(response.add_attribute("creator", creator));
+            }
+        }
+    }
+    Ok(response)
+}
+
+pub fn migrate_minter(
+    storage: &mut dyn Storage,
+    api: &dyn Api,
+    _env: &Env,
+    msg: &Cw721MigrateMsg,
+    response: Response,
+) -> StdResult<Response> {
+    match msg {
+        Cw721MigrateMsg::WithUpdate { minter, .. } => {
+            if let Some(minter) = minter {
+                MINTER.initialize_owner(storage, api, Some(minter.as_str()))?;
+                return Ok(response.add_attribute("creator", minter));
+            }
+        }
+    }
+    Ok(response)
+}
+
+/// Migrates only in case ownership is not present
+/// !!! Important note here: !!!
+/// - creator owns the contract and can update collection info
+/// - minter can mint new tokens
+///
+/// Before v0.19.0 there were confusing naming conventions:
+/// - v0.17.0: minter was replaced by cw_ownable, as a result minter is owner
+/// - v0.16.0 and below: minter was stored in dedicated `minter` store (so NOT using cw_ownable at all)
+pub fn migrate_legacy_minter_and_creator(
+    storage: &mut dyn Storage,
+    api: &dyn Api,
+    _env: &Env,
+    _msg: &Cw721MigrateMsg,
+    response: Response,
+) -> Result<Response, Cw721ContractError> {
+    let minter = MINTER.item.may_load(storage)?;
+    // no migration in case minter is already set
+    if minter.is_some() {
+        return Ok(response);
+    }
```

Comment:
Please note: new CREATOR uses `cw-ownable` singleton, which was before used by legacy minter. So new MINTER has new key. If new minter is set, then no need to migrate legacy! If not set, then ...